go/mcap/indexed_message_iterator.go

```diff
@@ -212,22 +211,29 @@ func (it *indexedMessageIterator) Next(p []byte) (*Schema, *Channel, *Message, e
 			return nil, nil, nil, err
 		}
 	}
-	for it.messageOffsetIdx == len(it.messageOffsets) {
-		err := it.loadNextChunk()
+
+	for it.indexHeap.Len() > 0 {
+		ri, err := it.indexHeap.HeapPop()
 		if err != nil {
 			return nil, nil, nil, err
 		}
+		if ri.messageIndexEntry == nil {
+			err := it.loadChunk(ri.chunkIndex)
+			if err != nil {
+				return nil, nil, nil, err
+			}
```

Comment:
I think we can eliminate a level of nesting here with a `continue`. Then the following `else` can be denested.